lib/utility/src/WatchdogWrapper.cpp

```diff
@@ -0,0 +1,35 @@
+
+#include "WatchdogWrapper.h"
+
+#include <string>
+
+#include "Logger.h"
+#include "ResetReason.h"
+#include "mbed.h"
+
+namespace Utility {
+
+void WatchdogWrapper::startWatchdog(std::chrono::milliseconds countdown_ms /*= 5000ms*/,
+                                    std::chrono::milliseconds pet_ms /*= 1000ms*/) {
+  uint32_t countdown_uint32 = countdown_ms.count();
+  Watchdog &watchdog        = Watchdog::get_instance();
+  watchdog.start(countdown_uint32);
+  Thread pet_thread;
+  pet_thread.start(callback(WatchdogWrapper::petWatchdog, &pet_ms));
+}
+
+void WatchdogWrapper::logResetReason() {
+  const reset_reason_t reason = ResetReason::get();
+  if (reason == RESET_REASON_WATCHDOG) {
+    time_t seconds = time(NULL);
+    Utility::logger << "Reset Reason: Watchog Timer - " << ctime(&seconds) << "\r\n";
```

Comment:
I think this time will initialize to Jan 1, 1970
https://os.mbed.com/docs/mbed-os/v6.10/apis/time.html

Reply:
Yes, i think we need to pass a value to set_time(), but since it will obviously be a different time everytime the function is run Im not sure how to handle this